src/ffi/c.rs

```diff
@@ -75,54 +87,65 @@ impl Drop for StreamWrapper {
     }
 }
 
-const ALIGN: usize = std::mem::align_of::<usize>();
+#[cfg(all(feature = "any_zlib", not(feature = "libz-rs-sys")))]
+mod allocator {
+    use super::*;
 
-fn align_up(size: usize, align: usize) -> usize {
-    (size + align - 1) & !(align - 1)
-}
+    use std::alloc::{self, Layout};
+    use std::convert::TryFrom;
+    use std::os::raw::c_void;
+
+    const ALIGN: usize = std::mem::align_of::<usize>();
 
-extern "C" fn zalloc(_ptr: *mut c_void, items: AllocSize, item_size: AllocSize) -> *mut c_void {
-    // We need to multiply `items` and `item_size` to get the actual desired
-    // allocation size. Since `zfree` doesn't receive a size argument we
-    // also need to allocate space for a `usize` as a header so we can store
-    // how large the allocation is to deallocate later.
-    let size = match items
-        .checked_mul(item_size)
-        .and_then(|i| usize::try_from(i).ok())
-        .map(|size| align_up(size, ALIGN))
-        .and_then(|i| i.checked_add(std::mem::size_of::<usize>()))
-    {
-        Some(i) => i,
-        None => return ptr::null_mut(),
-    };
-
-    // Make sure the `size` isn't too big to fail `Layout`'s restrictions
-    let layout = match Layout::from_size_align(size, ALIGN) {
-        Ok(layout) => layout,
-        Err(_) => return ptr::null_mut(),
-    };
-
-    unsafe {
-        // Allocate the data, and if successful store the size we allocated
-        // at the beginning and then return an offset pointer.
-        let ptr = alloc::alloc(layout) as *mut usize;
-        if ptr.is_null() {
-            return ptr as *mut c_void;
+    fn align_up(size: usize, align: usize) -> usize {
+        (size + align - 1) & !(align - 1)
+    }
+
+    pub type AllocSize = libz::uInt;
+
+    extern "C" fn zalloc(_ptr: *mut c_void, items: AllocSize, item_size: AllocSize) -> *mut c_void {
```

Comment:
The only two occurences of this `AllocSize` type are in the signature of `zalloc`, which is a private function. Can this type alias be inlined into the signature?